### PR TITLE
Expand reminders, allow DM notifications, listing and clearing

### PR DESCRIPTION
**Pull request by weeryan17**
- [x] Completed PR - Has all features or fixes that it was meant to and isn't half done
- [x] Follows the style guide - Style guide found [here](https://github.com/FlareBot/FlareBot/blob/master/CONTRIBUTING.md)
- [x] All changes have been tested

## Added
Dm remind option
Remind list
Remind clear

### CODE REVIEW

**Comment by WalshyDev** [comment]
Could we do some of the other requested features for reminders? `_r clear` to clear all reminders, `_r list` to list all reminders. I don't think it's worth removing specific ones as someone shouldn't have many and it makes it more complex on both ends. Also can we add a mention to the initial "reminder set"

**Comment by weeryan17** [comment]
@WalshyDev are we currently storing user reminders within the program anywhere?

**Comment by WalshyDev** [comment]
I don't think so... Flare2 OP

**Comment by weeryan17** [comment]
lol. so do you want me to make a system for that or wait till flare 2 for that and merge this now?

**Comment by WalshyDev** [comment]
Err... I think they could still be cleared.

**Comment by weeryan17** [comment]
that would still require me storing unless i wanted to mess around with cassandra and that type of stuff

**Comment by WalshyDev** [comment]
When the future tasks initially load, you can put them into a Set rather than just queueing them. That way you can access the object, also make sure to remove from the map when the time is up.

**Comment by weeryan17** [comment]
Do we want to store all future actions like that or just reminders?

**Comment by WalshyDev** [comment]
We may as well store all, allows for future changes.

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 1]
```diff
@@ -410,7 +414,9 @@ public JDA getClient() {
         return shardManager.getShards().get(0);
     }
 
+
     private void loadFutureTasks() {
+        futureActions = new ConcurrentHashSet<>();
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: No point even initialising down here. Just make a final set up there.

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 2]
```diff
@@ -383,6 +387,8 @@ protected void run() {
                 "Executed creations!", "Failed to execute creations!",
                 () -> manager.executeCreations());
 
+        futureActions = new ConcurrentHashSet<>();
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: Not here 🤦‍♂️ 
I said final set for a reason. Put it where the declaration is.

Comment (weeryan17) [reply src/main/java/stream/flarebot/flarebot/FlareBot.java]: I'm a derp

**Comment by weeryan17** [comment]
@WalshyDev while your here request a review from @binaryoverload please

**Comment by WalshyDev** [comment]
@binaryoverload re-review.

File: src/main/java/stream/flarebot/flarebot/commands/useful/RemindCommand.java [hunk 3]
```diff
@@ -13,22 +14,68 @@
...
+                } else if(args[0].equalsIgnoreCase("clear")) {
+                    Set<FutureAction> futureActions = FlareBot.instance().getFutureActions();
+                    for (FutureAction action : futureActions) {
+                        if (action.getAction().equals(FutureAction.Action.REMINDER) || action.getAction().equals(FutureAction.Action.DM_REMINDER)) {
+                            if (action.getResponsible() == sender.getIdLong()) {
+                                action.delete();
+                            }
+                        }
+                    }
+                    MessageUtils.sendSuccessMessage("Cleared your reminds successfully", channel, sender);
```
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/useful/RemindCommand.java]: reminders*

File: src/main/java/stream/flarebot/flarebot/commands/useful/RemindCommand.java [hunk 4]
```diff
@@ -44,7 +91,10 @@ public String getDescription() {
 
     @Override
     public String getUsage() {
-        return "`{%}remind <duration> <reminder>` - Reminds a user about something after a duration.";
+        return "`{%}remind <duration> <reminder>` - Reminds a user about something after a duration.\n" +
+                "`{%}remind <duration> dm <reminder>` - Reminds a user about something after a duration via Direct Messages.\n" +
+                "`{%}remind list` - Lists your current reminders.\n" +
+                "`{%}remind clear` - Clears your current reminds.";
```
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/useful/RemindCommand.java]: reminders*

**Pull request merged**